src/richie/apps/courses/models/course.py

```diff
@@ -484,6 +494,91 @@ def get_course_runs_dict(self):
 
         return dict(course_runs_dict)
 
+    def get_pace(self):
+        """
+        Returns the course pace computed according to duration and effort fields or
+        None if one of this field is blank or if `is_self_paced` flag is equal to False.
+
+        This method returns a tuple of three values:
+        - pace : the float value of the pace (ratio between effort / duration)
+        - pace_unit
+        - pace_reference_unit
+        """
+        if self.is_self_paced is True or self.duration is None:
+            return None
+
+        if self.effort is None:
+            raise ValueError("Cannot compute pace without effort.")
+
+        (effort, effort_unit) = self.effort
+        (duration, duration_unit) = self.duration
+
+        time_units_keys = list(dict(defaults.TIME_UNITS))
+        if time_units_keys.index(effort_unit) >= time_units_keys.index(duration_unit):
+            raise ValueError(
+                (
+                    "Cannot compute pace with effort unit less or equal than duration unit."
+                    f" ({effort_unit}/{duration_unit})"
+                )
+            )
+
+        pace = effort / duration
+        return (pace, effort_unit, duration_unit)
+
+    def get_pace_display(self):
+        """
+        Returns a human readable pace value to display.
+        If the course is self paced or has no duration, it returns `Self paced`.
+        If pace cannot be compute, it returns None.
+
+        To improve readability, we manage three cases with different renders:
+        - Pace is a full hour -> ~2 hours/day
+        - Pace is not a full hour -> ~2h30/day
+        - Pace is under a hour -> ~45 minutes/day
+
+        Furthermore, as this information is an estimation, we do not want value too precise,
+        so that's why we round minutes by fifteen minutes.
+        """
+
+        time_units = defaults.TIME_UNITS
+
+        try:
+            (pace, pace_unit, pace_reference_unit) = self.get_pace()
```

Comment:
you should only wrap this line in the try/except. Wrapping everything is an anti-pattern. It impacts readability and may hide some other errors than those you were targetting.